### ISSUE TITLE
Name a collection of columns by specifying just the first and the last column

### ISSUE THREAD

**Issue by FarrelBuch** [Rdatatable/data.table#1039]
So lets say one has a data.table with the following columns  

first.name, last.name, height, weight, shoe.size, eye.color, hair.length, appendage.size, ear.length

If one wanted to just include weight through hair.length one would have to go something such as this

dt[,list(weight, shoe.size, eye.color, hair.length)]

*Is there a way to do something along the lines of*

dt[,list(weight...hair.length)]

`base::subset()` does this

**Comment by DavidArenburg** [comment]
Maybe `dt[, 4:7, with = FALSE]`

**Comment by FarrelBuch** [comment]
@DavidArenburg I like the succinctness and simplicity but somewhere along the line when one uses script over and over again the table has a way of changing and that introduces columns which screws with the column number notation. I would prefer to name the first and last column of the range one is interested in.

**Comment by ChristK** [comment]
So why not...
````R
dt[, grep("weight", colnames(dt)):grep("hair.length", colnames(dt)), with = FALSE]
````
That should work even if columns added or removed before or after the 4 columns of interest. And you can right a function to make it look less ugly...

**Comment by arunsrinivasan** [comment]
Seems like this is a duplicate of #748. It's for `.SDcols`, but I'll expand it to other scenarios.

**Comment by arunsrinivasan** [comment]
@FarrelBuch, now implemented in b0ecc0e56850a543cef874106fcda11f7fafc6ac. Let me know if there're any issues.
